app/Http/Controllers/Auth/LoginController.php

```diff
@@ -37,4 +39,41 @@ public function __construct()
     {
         $this->middleware('guest')->except('logout');
     }
+
+    protected function redirectTo()
+    {
+        if (Auth()->user()->role_id == config('auth.roles.admin')) {
+            return route('admin');
+        } else {
+            return route('home');
+        }
+    }
+
+    public function login(Request $request)
+    {
+        $request->validate([
+            'email' => 'required',
+            'password' => 'required',
+        ]);
+        
+
+        $compare = Auth()->attempt([
```

Comment:
```suggestion
        $request->validate([
            'email' => 'required',
            'password' => 'required',
        ]);

        $compare = Auth()->attempt([
```
cách nhiều dòng quá bn, chuyển sang dùng Form Request nữa nha, bổ sung min,max length cho password, email nữa nha.